src/handlers/xp.js

```diff
@@ -0,0 +1,110 @@
+const sc = require('sourcecred').default
+const fetch = require('node-fetch')
+const { log } = require('../utils')
+const parseMyCred = require('../parser/myCred')
+const { environment } = require('../environment')
+
+module.exports = async function score(message) {
+  const NodeAddress = sc.core.address.makeAddressModule({
+    name: 'NodeAddress',
+    nonce: 'N',
+    otherNonces: new Map().set('E', 'EdgeAddress'),
+  })
+
+  const targetParameter = parseMyCred(message.content)
+  const targetUserDiscordID = targetParameter.slice(3, targetParameter.length - 1)
+
+  if (isNaN(targetUserDiscordID)) {
+    return message.reply(
+      'command parsing failed. Please use the !ac help command to see how to use the requested command properly.',
+    )
+  }
+
+  const credAccounts = await (
+    await fetch(`https://raw.githubusercontent.com/${environment('GITHUB_LEDGER_FILE_PATH')}`,
+    )
+  ).json()
+
+  try {
+    const accounts = credAccounts.accounts
+    for (let i = 0; i < accounts.length; i++) {
+      if (accounts[i].account.identity.subtype !== 'USER') continue
+
+      const discordAliases = accounts[i].account.identity.aliases.filter(
+        alias => {
+          const parts = NodeAddress.toParts(alias.address)
+          return parts.indexOf('discord') > 0
+        },
+      )
+      //no user on discord
+      if (!discordAliases.length) continue
+
+      let userTotalCred = 0
+      let lengthArray = 0
+      let userWeeklyCred = 0
+      let found = false
+
+      discordAliases.forEach(alias => {
+        const discordId = NodeAddress.toParts(alias.address)[4]
+        userTotalCred = accounts[i].totalCred
+        lengthArray = accounts[i].cred.length
+        userWeeklyCred = accounts[i].cred
+        if (discordId === targetUserDiscordID) {
+          found = true
```

Comment:
No need to use a flag like this, find the account with targetDiscordId first before you do the rest of the logic

Reply:
Updated as well!